zypp/base/Gettext.h

```diff
@@ -14,6 +14,17 @@
 #ifndef ZYPP_BASE_GETTEXT_H
 #define ZYPP_BASE_GETTEXT_H
 
+/** Undef translation macros if they already exist */
```

Comment:
IMO we should differ between the `#ifdef ZYPP_DLL` and `#else` case below.

- `#define N_(MSG)` should be moved into both cases

then

- In `#ifdef ZYPP_DLL`  we can unconditionally `#undef` each macro before `#define`ing it
- In the `#else` case we may define them if they are not already defined (by the application).